client/allocrunner/consulsock_hook.go

```diff
@@ -159,7 +159,12 @@ func (s *sockProxy) run(alloc *structs.Allocation) error {
 	hostGRPCSockPath := filepath.Join(s.allocDir.AllocDir, allocdir.AllocGRPCSocket)
 	listener, err := net.Listen("unix", hostGRPCSockPath)
 	if err != nil {
-		return fmt.Errorf("unable to create unix socket for Consul gRPC endpoint: %v", err)
+		os.Remove(hostGRPCSockPath)
```

Comment:
Note to reviewers: I'm not super-wild about this but I'm not catching the error from this because I want the `net.Listen` error to be what bubbles up to the user. Open to suggestions there.

Reply:
Perhapse we check to see if the socket is found prior to calling `net.Listen`. If so, remove it since we're the creator if it anyways.

Reply:
I was worried about time-of-check/time-of-use there. But given that this method is marked "not safe for concurrent use" I don't think that's a problem. Will do.